pkg/gardenlet/controller/shoot/seed_registration_control.go

```diff
@@ -339,6 +339,7 @@ func prepareSeedConfig(ctx context.Context, gardenClient client.Client, seedClie
 		},
 		DNS: gardencorev1beta1.SeedDNS{
 			IngressDomain: fmt.Sprintf("%s.%s", common.IngressPrefix, *(shoot.Spec.DNS.Domain)),
+			NginxIngress:  shootedSeedConfig.NginxIngressSettings,
```

Comment:
Should we try to first take the settings from the shoot's `.spec.dns` section? Only if required fields are missing here (e.g., `.spec.dns.providers[] ` is empty`, then we need the additional information from the shooted seed configuration).